cmd/minikube/cmd/config/config.go

```diff
@@ -34,81 +34,96 @@ type configFile interface {
 }
 
 type setFn func(string, string) error
-type MinikubeConfig map[string]interface{}
 
 type Setting struct {
 	name        string
-	set         func(MinikubeConfig, string, string) error
+	set         func(pkgConfig.MinikubeConfig, string, string) error
 	validations []setFn
 	callbacks   []setFn
 }
 
 // These are all the settings that are configurable
 // and their validation and callback fn run on Set
-var settings = []Setting{
-	{
-		name:        "vm-driver",
-		set:         SetString,
-		validations: []setFn{IsValidDriver},
-		callbacks:   []setFn{RequiresRestartMsg},
-	},
-	{
-		name:        "v",
-		set:         SetInt,
-		validations: []setFn{IsPositive},
-	},
-	{
-		name:        "cpus",
-		set:         SetInt,
-		validations: []setFn{IsPositive},
-		callbacks:   []setFn{RequiresRestartMsg},
-	},
-	{
-		name:        "disk-size",
-		set:         SetString,
-		validations: []setFn{IsValidDiskSize},
-		callbacks:   []setFn{RequiresRestartMsg},
-	},
-	{
-		name:        "host-only-cidr",
-		set:         SetString,
-		validations: []setFn{IsValidCIDR},
-	},
-	{
-		name:        "memory",
-		set:         SetInt,
-		validations: []setFn{IsPositive},
-		callbacks:   []setFn{RequiresRestartMsg},
-	},
-	{
-		name: "show-libmachine-logs",
-		set:  SetBool,
-	},
-	{
-		name:        "log_dir",
-		set:         SetString,
-		validations: []setFn{IsValidPath},
-	},
-	{
-		name: "kubernetes-version",
-		set:  SetString,
-	},
-	{
-		name: config.WantUpdateNotification,
-		set:  SetBool,
-	},
-	{
-		name: config.ReminderWaitPeriodInHours,
-		set:  SetInt,
-	},
-	{
-		name: config.WantReportError,
-		set:  SetBool,
-	},
-	{
-		name: config.WantReportErrorPrompt,
-		set:  SetBool,
-	},
+var settings []Setting
+
+func init() {
```

Comment:
No.  I  was considering programatically adding things to the settings, (for example all the constants in pkg/minikube/config/config.go and all of the addons in pkg/assets/addons.go.  I am not sure of the cons of using an init() though.